### ISSUE TITLE
Split off general dihedral rotation functionality from Action_MakeStructure

### ISSUE THREAD

**Issue by drroe** [Amber-MD/cpptraj#36]
Create new action like 'rotatedihedral' with simplified syntax for rotating dihedral angles.

**Comment by hainm** [comment]
this action would save lots of time and we don't need to write a tons of temp output files from tleap.

http://ambermd.org/tutorials/advanced/tutorial23/paramfit_1.html

>This script creates a random structure set using leap and saves each structure individually in a created directory. Then, it uses cpptraj to collate the structures together into a coordinate file.

**Comment by drroe** [comment]
dihedralscan random could be adapted to do this (or maybe it already can?)

**Comment by hainm** [comment]
I have not played with `dihedralscan` yet (since I don't know how to use it in `pytraj` + I prefer to see example first).

I just look at manual, it's likely an action to create random structure (which is really cool to me).

**Comment by drroe** [comment]
From my extended test suite, 2 examples:
```
trajin ../tz2.rst7
# Rotate all phi psi dihedrals in intervals of 120 deg.
dihedralscan interval -120 outtraj rotations.mdcrd phi psi
```
And
```
trajin ../tz2.rst7
# Rotate dihedrals randomly; maxfactor relates to how many tries to attempt
dihedralscan random rseed 1 check maxfactor 10 phi psi
trajout random.mol2
```
Again, not sure how robust the code is, it's been a while since I used it. The manual has explanation of the keywords (I think).

**Comment by hainm** [comment]
Thanks. I will try with pytraj.

On Wednesday, August 5, 2015, Daniel R. Roe <notifications@github.com>
wrote:

> From my extended test suite, 2 examples:
>
> trajin ../tz2.rst7
> # Rotate all phi psi dihedrals in intervals of 120 deg.
> dihedralscan interval -120 outtraj rotations.mdcrd phi psi
>
> And
>
> trajin ../tz2.rst7
> # Rotate dihedrals randomly; maxfactor relates to how many tries to attempt
> dihedralscan random rseed 1 check maxfactor 10 phi psi
> trajout random.mol2
>
> Again, not sure how robust the code is, it's been a while since I used it.
> The manual has explanation of the keywords (I think).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Amber-MD/cpptraj/issues/36#issuecomment-128173227>.
>


-- 
Hai Nguyen

**Comment by hainm** [comment]
@drroe  

I just tried with `pytraj` and the action work fine (not sure if it is robust or not, at least I can see the animation :D). 

Just want to make sure: cpptraj only write traj to disk and not dump new traj to dataset, right?

Right now I don't know how to name the method: already have `make_structure`, 'make_random_structure' seems not to be correct.

**Comment by drroe** [comment]
Addressed by #382.
